server/app/controllers/MaintenanceController.scala

```diff
@@ -1,15 +1,53 @@
 package controllers
 
+import com.alexitc.playsonify.models.pagination.Limit
+import com.xsn.explorer.data.async.{BlockFutureDataHandler, TransactionFutureDataHandler}
+import com.xsn.explorer.gcs.GolombEncoding
+import com.alexitc.playsonify.core.FutureOr.Implicits.FutureOps
+import com.xsn.explorer.data.anorm.AnormPostgresDataHandler
+import com.xsn.explorer.data.anorm.dao.BlockFilterPostgresDAO
+import com.xsn.explorer.models.values.Height
 import controllers.common.{MyJsonController, MyJsonControllerComponents}
 import javax.inject.Inject
 import org.scalactic.Good
+import play.api.db.Database
 import play.api.libs.json.JsObject
 
 import scala.concurrent.Future
 
-class MaintenanceController @Inject()(components: MyJsonControllerComponents) extends MyJsonController(components) {
+class MaintenanceController @Inject()(
+    override val database: Database,
+    blockDataHandler: BlockFutureDataHandler,
+    transactionDataHandler: TransactionFutureDataHandler,
+    blockFilterPostgresDAO: BlockFilterPostgresDAO,
+    components: MyJsonControllerComponents
+) extends MyJsonController(components)
+    with AnormPostgresDataHandler {
 
   def run(query: String) = public { _ =>
+    val startingHeight = query.toInt
+
+    (startingHeight to 0 by -1).foreach(height => {
+      for {
+        block <- blockDataHandler.getBy(Height(height)).toFutureOr
+        transactions <- transactionDataHandler.getTransactionsWithIOBy(block.hash, Limit(100), None).toFutureOr
```

Comment:
the filter will be incorrect on blocks with more than 100 transactions, use a high number like 100k.